src/tests/test_opaque_vectors.rs

```diff
@@ -238,6 +246,45 @@ fn tests() -> Result<(), ProtocolError> {
         test_fake_vectors::<Ristretto255Sha512NoKsf>(&ristretto_fake_tvs)?;
     }
 
+    #[cfg(feature = "curve25519")]
+    {
+        struct Ristretto255Sha512Curve25519NoKsf;
+        impl CipherSuite for Ristretto255Sha512Curve25519NoKsf {
+            type OprfCs = crate::Ristretto255;
+            type KeGroup = crate::Curve25519;
+            type KeyExchange = TripleDh;
+            type Ksf = Identity;
+        }
+
+        let ristretto_real_tvs = json_to_test_vectors!(
+            rfc,
+            "Real",
+            "ristretto255-SHA512, x25519",
+            Ristretto255Sha512Curve25519NoKsf
+        );
+
+        let ristretto_fake_tvs = json_to_test_vectors!(
+            rfc,
+            "Fake",
+            "ristretto255-SHA512, x25519",
+            Ristretto255Sha512Curve25519NoKsf
+        );
+
+        assert!(
+            !(ristretto_real_tvs.is_empty() || ristretto_fake_tvs.is_empty()),
+            "Parsing error"
+        );
+
+        test_registration_request::<Ristretto255Sha512Curve25519NoKsf>(&ristretto_real_tvs)?;
+        test_registration_response::<Ristretto255Sha512Curve25519NoKsf>(&ristretto_real_tvs)?;
+        //test_registration_upload::<Ristretto255Sha512Curve25519NoKsf>(&ristretto_real_tvs)?;
+        //test_ke1::<Ristretto255Sha512Curve25519NoKsf>(&ristretto_real_tvs)?;
+        //test_ke2::<Ristretto255Sha512Curve25519NoKsf>(&ristretto_real_tvs)?;
+        //test_ke3::<Ristretto255Sha512Curve25519NoKsf>(&ristretto_real_tvs)?;
+        //test_server_login_finish::<Ristretto255Sha512Curve25519NoKsf>(&ristretto_real_tvs)?;
+        //test_fake_vectors::<Ristretto255Sha512Curve25519NoKsf>(&ristretto_fake_tvs)?;
```

Comment:
These are the tests that aren't working.

Reply:
I did try removing the clamping (https://github.com/cfrg/draft-irtf-cfrg-opaque/pull/404#discussion_r1145556262) from the random scalar generation but still couldn't produce the same output.

Reply:
A couple of things that I'm noticing (in-progress):
- The generated test vectors still seem to be producing private keys that are not clamped. This is an issue with how they are being generated, and am investigating in the spec PR. Once this is addressed, it should fix the test_ke1 tests.
- We need to adjust the diffie-hellman() function to compute the x25519 function, instead of just multiplying "our" sk with "their" pk. Once this is addressed, it should fix the test_ke2 and test_ke3 tests.

Reply:
> * We need to adjust the diffie-hellman() function to compute the x25519 function (https://docs.rs/x25519-dalek/latest/x25519_dalek/fn.x25519.html), instead of just multiplying "our" sk with "their" pk.

Unless I'm missing anything the underlying code only does the multiplication too: https://docs.rs/x25519-dalek/2.0.0-pre.1/src/x25519_dalek/x25519.rs.html#317-319.

Reply:
Oh right I see, perhaps that part should be fine then :) But yes currently we are waiting on a re-generation of the private keys to ensure that they are clamped.